lib/Db/AliasMapper.php

```diff
@@ -64,4 +66,24 @@ public function findAll(int $accountId, string $currentUserId): array {
 
 		return $this->findEntities($qb);
 	}
+
+	/**
+	 * @param int $accountId the account whose aliases will be deleted
+	 * @param string $currentUserId the user that is currently logged in
+	 */
+	public function deleteAll($accountId, $currentUserId) {
+		$qb = $this->db->getQueryBuilder();
+
+		$query = $qb->delete($this->getTableName(), 'aliases')
+			->join('aliases', 'mail_accounts', 'accounts', $qb->expr()->eq('aliases.account_id', 'accounts.id'))
+			->where($qb->expr()->eq('account_id', $qb->createNamedParameter($accountId)))
+			->where(
```

Comment:
andWhere?

Reply:
What a suspense!

Reply:
Ah nooo, it continues after. I'm disappointed, I thought it was a cliffhanger! :grin: